frontend/awx/access/credentials/CredentialForm.tsx

```diff
@@ -240,8 +305,40 @@ export function EditCredential() {
       }
     });
     const modifiedCredential = { ...editedCredential, inputs: pluginInputs };
-    await patch(awxAPI`/credentials/${id.toString()}/`, modifiedCredential);
-    navigate(-1);
+    const credentialInputSourcePayload = accumulatedPluginValues.map((credentialInputSource) => ({
+      ...credentialInputSource,
+      target_credential: credential?.id,
+    }));
+
+    if (pluginsToDeletePayload && pluginsToDeletePayload.length > 0) {
+      await Promise.all(
+        pluginsToDeletePayload.map(async (id) => {
+          await deleteRequest(awxAPI`/credential_input_sources/${id.toString()}/`);
+        })
+      ).then(async () => {
+        await patch(awxAPI`/credentials/${id.toString()}/`, modifiedCredential);
+        await Promise.all(
+          credentialInputSourcePayload.map(async (credentialInputSource) => {
+            await postRequest(
```

Comment:
Could use we `usePostRequest` here? usePostRequest takes care of cleaning the cache for a specific URL.